src/System.Runtime.Extensions/tests/System/Convert.BoxedObjectCheck.cs

```diff
@@ -9,33 +10,69 @@ namespace System.Tests
-            UInt16.MinValue, UInt16.MaxValue,
-            Int32.MinValue, Int32.MaxValue, (Int32)0,
-            UInt32.MinValue, UInt32.MaxValue,
-            Int64.MinValue, Int64.MaxValue, (Int64)0,
-            UInt64.MinValue, UInt64.MaxValue,
-            Char.MinValue, Char.MaxValue, (Char)0,
-            Double.MinValue, Double.MaxValue, (Double)0,
-            Single.MinValue, Single.MaxValue, (Single)0,
-            Decimal.MinValue, Decimal.MaxValue, (Decimal)0,
-            DateTime.MinValue, DateTime.Now, DateTime.MaxValue
-        };
+            yield return new object[] { true };
+            yield return new object[] { false };
+            yield return new object[] { Byte.MinValue };
+            yield return new object[] { Byte.MaxValue };
+            yield return new object[] { SByte.MinValue };
+            yield return new object[] { SByte.MaxValue };
+            yield return new object[] { (SByte)0 };
+            yield return new object[] { Int16.MinValue };
+            yield return new object[] { Int16.MaxValue };
+            yield return new object[] { (Int16)0 };
+            yield return new object[] { UInt16.MinValue };
+            yield return new object[] { UInt16.MaxValue };
+            yield return new object[] { Int32.MinValue };
+            yield return new object[] { Int32.MaxValue };
+            yield return new object[] { (Int32)0 };
+            yield return new object[] { UInt32.MinValue };
+            yield return new object[] { UInt32.MaxValue };
+            yield return new object[] { Int64.MinValue };
+            yield return new object[] { Int64.MaxValue };
+            yield return new object[] { (Int64)0 };
+            yield return new object[] { UInt64.MinValue };
+            yield return new object[] { UInt64.MaxValue };
+            yield return new object[] { Char.MinValue };
+            yield return new object[] { Char.MaxValue };
+            yield return new object[] { (Char)0 };
+            yield return new object[] { Double.MinValue };
+            yield return new object[] { Double.MaxValue };
+            yield return new object[] { (Double)0 };
+            yield return new object[] { Single.MinValue };
+            yield return new object[] { Single.MaxValue };
+            yield return new object[] { (Single)0 };
+            yield return new object[] { Decimal.MinValue };
+            yield return new object[] { Decimal.MaxValue };
+            yield return new object[] { (Decimal)0 };
+            yield return new object[] { DateTime.MinValue };
+            yield return new object[] { DateTime.Now };
+            yield return new object[] { DateTime.MaxValue };
+        }
 
-            foreach (object obj in testValues)
+        [Theory]
+        [MemberData(nameof(DefaultToTypeValues))]
+        public static void TestConvertedCopies(object testValue)
+        {
+            Assert.All(DefaultToTypeValues(), input =>
```

Comment:
I think the idea is that it's too much work / too complicated to try to create all of the valid combinations of convertible types, so we just try them all and ignore the ones that don't work. Presumably we have other tests that ensure the correct type conversions work, so this seems OK to me.

Are we able to catch a more specific exception than this, at least?